app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,31 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @like = Like.new
+  end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new_post, locals: { post: } }
+    end
+  end
+
+  def create
+    post = Post.new(title: params[:post][:title], text: params[:post][:text], user: current_user)
+
+    if post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to "/users/#{current_user.id}/posts"
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render inline: '<p>Error: Post could not be saved<p>'
+    end
+  end
+
+  def like
```

Comment:
- [ ] @for-ashraf great work adding a like method in your post controller, it's not good practice to add the way you did, so kindly create a like controller to handle the like functionality (it's best practice to separate the functionality of our application for easier scalability of the app in future).